release-tools/release_image_github.sh

```diff
@@ -0,0 +1,158 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+script_name="${0##*/}"
+script_dir=$(dirname $(realpath -s "$0"))
+
+tmp_dir=$(mktemp -t -d cc-bump.XXXX)
+release_tool="${tmp_dir}/release-tool"
+osbuilder_dir="${tmp_dir}/osbuilder/"
+repo_owner="clearcontainers"
+push=false
+
+versions_file="https://raw.githubusercontent.com/clearcontainers/runtime/master/versions.txt"
+os_version="$(curl -Ls ${versions_file} | grep '^clear_vm_image_version=' | cut -d= -f 2)"
+agent_version="$(curl -Ls ${versions_file} | grep '^cc_agent_version=' | cut -d= -f 2)"
+
+function cleanup {
+	if [ -d "${osbuilder_dir}" ]; then
+		info "Remove temp dir: ${osbuilder_dir}"
+		sudo rm -rf "${osbuilder_dir}"
+	fi
+
+	info "Remove temp dir: ${tmp_dir}"
+	rm  -rf "${tmp_dir}"
+}
+
+trap cleanup EXIT
+
+die() {
+	echo >&2 -e "\e[1mERROR\e[0m: $*"
+	exit 1
+}
+
+info() {
+	echo -e "\e[1mINFO\e[0m: $*"
+}
+
+
+
+source "${script_dir}/../versions.txt"
+
+
+function build_image() {
+	export USE_DOCKER=1
+
+	make clean
+	AGENT_VERSION="${agent_version}" OS_VERSION="${os_version}" make rootfs
+	make image
+}
+
+build_release_tool() {
+	pushd "${script_dir}/../release-tools/"
+	info "build release tool"
+	go build -o  "${release_tool}"
+	popd
+}
+
+release_image() {
+	info "Creating image tarball"
+	image_version="cc-${os_version}-agent-${agent_version:0:6}"
+	dist_bin_name="image-${image_version}-binaries"
+	image_name="${image_version}.img"
+	mkdir -p "${dist_bin_name}"
+
+	cp workdir/container.img "${dist_bin_name}/${image_name}"
+	cp workdir/image_info "${dist_bin_name}/${image_version}.info"
+	sed \
+		-e "s|@IMAGE@|${image_name}|g" \
+		-e "s|@VMLINUX@|${dist_vmlinuz}|g" \
+		"${script_dir}/Makefile.dist.install.image" > "${dist_bin_name}/Makefile"
+
+	tarball="${dist_bin_name}.tar.gz"
+	tar -zvcf "${tarball}" "${dist_bin_name}"
+	shasum="SHA512SUMS"
+	sha512sum "${tarball}" > "${shasum}"
+
+	if [ -n "${output_dir}" ]; then 
+		[ -d "${output_dir}" ] ||  die "$output_dir not  a directory"
+		cp "${tarball}" "${output_dir}"
+		cp "${shasum}" "${output_dir}"
+	fi
+	if [ "${push}" = true ]; then
+		[ -z "$GITHUB_TOKEN" ] && die "\$GITHUB_TOKEN is empty"
+		build_release_tool
+		info "push changes to clearcontainers"
+		info "creating release"
+		set -x
+		${release_tool} --owner "${repo_owner}" \
+			release \
+			--asset "${tarball}" \
+			--asset "${shasum}" \
+			--force-version \
+			--version "${image_version}" "osbuilder"
+		set +x
+	fi
+
+}
+
+usage() { 
+cat << EOT
+Usage: $0 [options]
+Script to build and publish an new Clear Containers Image.
+
+Options:
+-h              : show this help
+-o <dir>        : Create resulting image in <dir>
+-p              : Push to git-repo default: ${push}
+-r <repo-owner> : git repository to push image
+-t <token>      : Github token to create new release. ENV: \$GITHUB_TOKEN
```

Comment:
which one have more priority?

Reply:
the -t options has more priority, added usage.